akka-cluster/src/main/scala/akka/cluster/protobuf/ClusterMessageSerializer.scala

```diff
@@ -18,72 +18,65 @@ import scala.collection.JavaConverters._
 import scala.concurrent.duration.Deadline
 import java.io.NotSerializableException
 
+import akka.annotation.InternalApi
 import akka.cluster.InternalClusterAction._
 import akka.cluster.routing.{ ClusterRouterPool, ClusterRouterPoolSettings }
 import akka.routing.Pool
 import com.typesafe.config.{ Config, ConfigFactory, ConfigRenderOptions }
 
 /**
- * Protobuf serializer of cluster messages.
+ * INTERNAL API
  */
-class ClusterMessageSerializer(val system: ExtendedActorSystem) extends BaseSerializer {
+@InternalApi
+private[akka] object ClusterMessageSerializer {
+  // FIXME use short manifests when we can break wire compatibility
+  // needs to be full class names for backwards compatibility
+  val JoinManifest = classOf[InternalClusterAction.Join].getName
```

Comment:
I wonder if we should replace them now with the current names?

Reply:
We can't because we need to be able to interact with (join) 2.5.9-nodes in a rolling upgrade.

Reply:
Or, you mean hardcode the names so we don't break if we refactor?

Reply:
Yep, hardcoding seems safer.

Reply:
God point. I'll do that.

Reply:
The rest of us try to make good points, but Johannes makes God points.